### ISSUE TITLE
Reduce_sum error

### ISSUE THREAD

**Issue by Lillypucien** [tensorflow/tensorflow#21439]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: YES
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 64 bits
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.9.0
- **Python version**: 3.5
- **CUDA/cuDNN version**: 9.0 / v7
- **GPU model and memory**: Nvidia GTX TITANX 12GB, driver up to date (398.82)
- **Bazel version**: N/A
- **Mobile device**: N/A
- **Exact command to reproduce**:

### Describe the problem
Hi,

See test case below. On this specific setting it happens that tf.reduce_sum returns wrong values, as attested by the loss which should always return 0, but does not in this case (see logs at the end).

The bug does not happen when running on CPU, or when batchSize is strictly lesser than 33, or when "models" tensor has only 3 dimensions (even if we increase its size), or when "models" is smaller, or when the reduce_sum is applied on all axes instead of the last axis.

The reduce_sum is also correct for the first 32 elements of its first dimension, and only goes wrong for the 33rd (and next) element(s). 

If we replace the "diff = inputTensor - inputTensor" tensor by a tf.zeros tensor, the bug does not appear.

The bug also happens with other reducing tensorflow methods (such as tf.reduce_min, tf.reduce_max and tf.reduce_mean).

Thanks in advance

### Source code / logs
```
import tensorflow as tf
import numpy as np
from random import random

batchSize = 33

np.random.seed(0)
models = np.random.rand(batchSize, 2050, 2050, 2)
models = np.asarray(models, dtype = "float32")

dataPlaceholder = tf.placeholder("float32", models.shape)
inputTensor = dataPlaceholder

diff = inputTensor - inputTensor
# diff = tf.zeros(inputTensor.shape)

sum = tf.reduce_sum(diff, axis = 3)

loss = tf.reduce_max(sum)
loss = tf.Print(loss, [diff[batchSize-1,0,0], sum[batchSize-1,0,0]], message="diff[batchSize-1,0,0], reduce_sum[batchSize-1,0,0] : ")

if __name__ == '__main__':
	with tf.Session() as sess:
		batchLoss = sess.run(loss, feed_dict = {dataPlaceholder: models[:batchSize]})
		print("batch loss : {}".format(batchLoss))

---------------------------------------------------
batch loss : 0.999999463558197
diff[batchSize-1,0,0], reduce_sum[batchSize-1,0,0] : [0 0][0.154408231]
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
Mobile device

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by harshini-gadige** [comment]
@rmlarsen Hi, could you please look into this issue ?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jvishnuvardhan** [comment]
@Lillypucien Looks like this was resolved in recent TF1.x. I tried your code with `TF1.15.3` and I cannot reproduce the issue. Please take a look the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/247c54a0df162eebec9513625add28d0/untitled.ipynb).

Please verify once and close the issue if this was resolved for you. Thanks!
